ewms_pilot/pilot.py

```diff
@@ -3,78 +3,161 @@
 
 import argparse
 import asyncio
+import enum
+import json
+import logging
 import pickle
 import subprocess
 import sys
 import time
 from pathlib import Path
-from typing import Any, Optional
+from typing import Any, Callable, Optional, TypeVar
 
 import asyncstdlib as asl
 from wipac_dev_tools import logging_tools
 
-from . import LOGGER, mq
-
-OUT_PKL = Path("out_msg.pkl")
-IN_PKL = Path("in_msg.pkl")
-
-
-def inmsg_to_infile(in_msg: Any, debug_infile: Optional[Path]) -> Path:
+from .mq import mq
+
+LOGGER = logging.getLogger("ewms-pilot")
+
+
+class FileType(enum.Enum):
+    """Various file types/extensions."""
+
+    PICKLE = "pkl"
+    PLAIN_TEXT = "txt"
+    JSON = "json"
+    BINARY = "bin"
+
+
+class UniversalFileInterface:
+    """Support reading and writing for any `FileType` file extension."""
+
+    @classmethod
+    def write(cls, in_msg: Any, fpath: Path) -> None:
+        """Write `stuff` to `fpath` per `fpath.suffix`."""
+        cls._write(in_msg, fpath)
+        LOGGER.info(f"File Written :: {fpath} ({fpath.stat().st_size} bytes)")
+
+    @classmethod
+    def _write(cls, in_msg: Any, fpath: Path) -> None:
+        LOGGER.info(f"Writing payload to file @ {fpath}")
+        LOGGER.debug(in_msg)
+
+        # PICKLE
+        if fpath.suffix == FileType.PICKLE.value:
+            with open(fpath, "wb") as f:
+                pickle.dump(in_msg, f)
+        # PLAIN_TEXT
+        elif fpath.suffix == FileType.PLAIN_TEXT.value:
+            with open(fpath, "w") as f:
+                f.write(in_msg)
+        # JSON
+        elif fpath.suffix == FileType.JSON.value:
+            with open(fpath, "w") as f:
+                json.dump(in_msg, f)
+        # BINARY
+        elif fpath.suffix == FileType.BINARY.value:
+            with open(fpath, "wb") as f:
+                f.write(in_msg)
+        # ???
+        else:
+            raise ValueError(f"Unsupported file type: {fpath.suffix} ({fpath})")
```

Comment:
Should this use the new `match`/`case` format?  Or does this need to be backwards compatible?

Reply:
(see above)